compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/NemoOptimizer.java

```diff
@@ -113,6 +124,24 @@ public IRDAG optimizeAtRunTime(final IRDAG dag, final Message message) {
     return optimizationPolicy.runRunTimeOptimizations(dag, message);
   }
 
+  /**
+   * Operations to be done prior to the Compile-Time Optimizations.
+   *
+   * @param dag    the DAG to process.
+   * @param policy the optimization policy to optimize the DAG with.
+   */
+  private void beforeCompileTimeOptimization(final IRDAG dag, final Policy policy) {
+    if (policy instanceof XGBoostPolicy) {
+      clientRPC.send(ControlMessage.DriverToClientMessage.newBuilder()
+        .setType(ControlMessage.DriverToClientMessageType.LaunchOptimization)
```

Comment:
Hmm.. it assumes that we already have enough metrics in DB...

Reply:
At least, we need to specify all assumptions.

Reply:
What happen if we have no metrics?

Reply:
Without enough metrics, it simply runs the DAG in with the DefaultPass, which i think is sensible.